psydac/api/grid.py

```diff
@@ -2,25 +2,71 @@
 
 import numpy as np
 
+from functools import reduce
+
 from psydac.core.bsplines          import find_span
 from psydac.core.bsplines          import basis_funs_all_ders
 from psydac.core.bsplines          import basis_ders_on_quad_grid
 from psydac.fem.splines            import SplineSpace
 from psydac.fem.tensor             import TensorFemSpace
 from psydac.fem.vector             import ProductFemSpace
+
+def get_points_weights(spaces, axis, e):
+    for s in spaces:
+        if e in s.quad_grids[axis].indices:
+            i = np.where(s.quad_grids[axis].indices==e)[0][0]
+            return s.quad_grids[axis].points[i:i+1], s.quad_grids[axis].weights[i:i+1]
 #==============================================================================
 class QuadratureGrid():
-    def __init__( self, V , axis=None, ext=None):
+    def __init__( self, V , axis=None, ext=None, trial_space=None):
+
+        n_elements          = []
+        indices             = []
```

Comment:
What is contained in the `indices` list?

Reply:
it represents the union of the indices calculated by `FemAssemblyGrid`, we make sure that they match between all the spaces.

Reply:
Please add a comment to the code!

Reply:
done